h/api/nipsa/search.py

```diff
@@ -0,0 +1,61 @@
+def nipsa_filter(userid=None):
+    """Return an Elasticsearch filter for filtering out NIPSA'd annotations.
+
+    The returned filter is suitable for inserting into an Es query dict.
+    For example:
+
+        query = {
+            "query": {
+                "filtered": {
+                    "filter": nipsa_filter(),
+                    "query": {...}
+                }
+            }
+        }
+
+    :param userid: The ID of a user whose annotations should not be filtered.
+        The returned filtered query won't filter out this user's annotations,
+        even if the annotations have the NIPSA flag.
+    :type userid: unicode
+
+    """
+    # If any one of these "should" clauses is true then the annotation will
+    # get through the filter.
+    should_clauses = [{"not": {"term": {"nipsa": True}}}]
+
+    if userid:
```

Comment:
So you're saying we assume the value is either `None` or a valid userid (so not the empty string and not falsy). In that case `if userid:` and `if userid is not None:` are the same behaviour-wise and it comes down to style. Our style guide says to use the implicit false _if at all possible_ and here it is possible to get the right behaviour by using it ... So unless we want to introduce a new style rule along the lines of always use `is None` instead of the implicit false, or always use it in case of `None` as a sentinel value, or something.